mesonbuild/linkers.py

```diff
@@ -1372,3 +1372,88 @@ def get_soname_args(self, env: 'Environment', prefix: str, shlib_name: str,
                         suffix: str, soversion: str, darwin_versions: T.Tuple[str, str],
                         is_shared_module: bool) -> T.List[str]:
         return []
+
+class TCCDynamicLinker(PosixDynamicLinkerMixin, DynamicLinker):
+    """Linker for TCC compiler."""
+    id = 'tcc'
+
+    def __init__(self, for_machine: mesonlib.MachineChoice,
+                 *, version: str = 'unknown version'):
+        super().__init__(['tcc'], for_machine, '-Wl,', [],
+                         version=version)
+
+    def get_allow_undefined_args(self) -> T.List[str]:
+        return []
+
+    def get_pic_args(self) -> T.List[str]:
+        return []
+
+    def get_link_whole_for(self, args: T.List[str]) -> T.List[str]:
+        if not args:
+            return args
+        return self._apply_prefix('--whole-archive') + args + self._apply_prefix('--no-whole-archive')
+
+    def get_coverage_args(self) -> T.List[str]:
+        return []
+
+    def export_dynamic_args(self, env: 'Environment') -> T.List[str]:
+        m = env.machines[self.for_machine]
+        if m.is_windows() or m.is_cygwin():
+            return self._apply_prefix('--export-all-symbols')
+        return self._apply_prefix('-export-dynamic')
```

Comment:
It seems that the tcc's equivalent of "-Wl,--export-dynamic" is "-rdynamic" option.

https://lists.gnu.org/archive/html/libtool-patches/2014-07/msg00001.html

Reply:
With tcc 0.9.27, it seems that '-Wl,--export-all-symbols' is the exact equivalent of '-rdynamic'.

Reply:
From man page:
```
-Wl,--export-all-symbols
-Wl,--export-dynamic
           Export global symbols to the dynamic linker. It is useful when a library opened with "dlopen()" needs to access executable symbols.

```
I'm not sure there is really a need to special case windows/cygwin, I probably copied that code from another linker around :)

Reply:
> I probably copied that code from another linker around
That's exactly the way I was doing the same thing. :-)